src/components/button/toggling-button.ts

```diff
@@ -0,0 +1,32 @@
+/**
+ * Create Toggling Button
+ * Adds an event listener to toggle the `aria-pressed` value
+ *
+ * @param togglingButton - The button to apply the toggling functionality to
+ */
+export const initTogglingButton = (togglingButton: HTMLButtonElement) => {
```

Comment:
Will it always be initialized with a `<button>` or are there cases where the server-rendered HTML could render on an `<a>` or another element?

Reply:
@calebeby I introduced [template logic with the intent to only render a `<button>` element](https://github.com/cloudfour/cloudfour.com-patterns/pull/1318/files#diff-7db3e7fdff7b358ebbc9d41aa97a09ae3e5d9ce13ed04e9d21e5f5c0a454fc6aR6) when the `aria_pressed` value is set. 

So in theory, there shouldn't be cases where an `<a>` or another element are rendered. 

Can you elaborate a bit more on your question? Did you spot an opportunity to improve the logic? 🔍 👀

Reply:
Got it! 👍 I wasn't sure if there were any cases where we would use an `<a>` for the toggle button but it sounds like there aren't.